### PR TITLE
Update react-scripts, TS and remove ESlint as it exist in CRA

### PR DESCRIPTION
**Pull request by lipis**

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -2,6 +2,9 @@
   "extends": ["prettier", "react-app"],
   "plugins": ["prettier"],
   "rules": {
+    "@typescript-eslint/no-unused-vars": "off",
+    "@typescript-eslint/no-redeclare": "off",
+    "import/no-anonymous-default-export": "off",
```
Comment (lipis) [inline .eslintrc.json:7]: Ideally would be nice to get rid of those.

**Comment by schuchard** [comment]
Sounds like a bug - https://github.com/facebook/create-react-app/issues/9429#issuecomment-674994709

> "noFallthroughCasesInSwitch": true to compilerOptions in my tsconfig.json

**Comment by lipis** [comment]
@schuchard There is a different error now.. It's a bit late here.. if anyone knows something, all ideas are welcome!

**Comment by schuchard** [comment]
looking at the issue I posted, a common thread with "working" repos is `typescript` `4.X.X`, this is still on `<4`. Might be worth a try?

**Comment by lipis** [comment]
@salehmoh Did more experiments.. still errors.

File: src/tests/regressionTests.test.tsx [hunk 2]
```diff
@@ -210,7 +210,7 @@ describe("regression tests", () => {
     mouse.up(10, 10);
 
     const transformHandles = getTransformHandles("mouse");
-    delete transformHandles.rotation; // exclude rotation handle
+    transformHandles.rotation = [0, 0];
```
Comment (lipis) [inline src/tests/regressionTests.test.tsx:213]: Not sure about this one.

**Comment by dwelle** [comment]
The issue is that `canvas.prototype.getContext("2d")` is returning `undefined`. It seems related to CRA, but I couldn't figure out the cause.

In clean jest project it works, with identical versions to CRA@4:

```
`-- jest@26.6.0
  `-- @jest/core@26.6.1
    `-- jest-config@26.6.1
      `-- jest-environment-jsdom@26.6.1
        `-- jsdom@16.4.0
```

Clean CRA@4 install:

```
`-- react-scripts@4.0.0
  `-- jest@26.6.0
    `-- @jest/core@26.6.1
      `-- jest-config@26.6.1
        `-- jest-environment-jsdom@26.6.1
          `-- jsdom@16.4.0
```

We can raise an issue in CRA repo, but in the meantime we'll probably have to revert to CRA@3.

**Comment by dwelle** [comment]
Getting closer: the canvas thing is caused by `resetMocks: true` introduced in https://github.com/facebook/create-react-app/pull/7899.

We still get failing snapshots, will investigate later.

**Comment by lipis** [comment]
Strange!

**Comment by dwelle** [comment]
This PR appears to have introduced snapshot regressions, but those were caused by previous dependency updates PRs. If you check the diff of [latest commit](ed4bf644b397f350785cf2c8ba56987377298983) (in this PR) and a [commit before we started updating deps](https://github.com/excalidraw/excalidraw/commit/9e40dcdcdce87ea8b00d5c021eed5097e1817207), it shows no snapshot changes: https://github.com/excalidraw/excalidraw/compare/9e40dcdcdce87ea8b00d5c021eed5097e1817207..ed4bf644b397f350785cf2c8ba56987377298983

**Pull request merged**